test/parallel/test-cluster-net-send.js

```diff
@@ -38,20 +34,20 @@ if (process.argv[2] !== 'child') {
       process.send('handle', socket);
   }
 
-  const server = net.createServer(function(c) {
-    process.once('message', common.mustCall(function(msg) {
+  const server = net.createServer((c) => {
+    process.once('message', common.mustCall((msg) => {
       assert.strictEqual(msg, 'got');
       c.end('hello');
     }));
     socketConnected();
   });
 
-  server.listen(common.PORT, function() {
-    socket = net.connect(common.PORT, '127.0.0.1', socketConnected);
+  server.listen(common.PORT, () => {
+    socket = net.connect(server.address().port, server.address().address,
```

Comment:
I think this is the same issue as the one I saw in https://github.com/nodejs/node/pull/10854

The problem is that you can't use `server.address().address` here. You should be listening on `0` (i.e. all IPv6 and IPv4 addresses), in which case `server.address().address` resolves to `::`. On linux this sometimes resolves to `127.0.0.1` if there is no IPv6 support on the machine, but this isn't the case on all platforms.

Changing to:

```diff
-  server.listen(common.PORT, () => {
-    socket = net.connect(server.address().port, server.address().address,
+  server.listen(0, () => {
+    socket = net.connect(server.address().port, 'localhost',
```

should fix the issue on all platforms. `localhost` should always resolve to either `127.0.0.1` or `::` (either of which is fine). 

cc/ @cjihrig